jina/peapods/peas/__init__.py

```diff
@@ -158,41 +152,18 @@ def __init__(self, args: 'argparse.Namespace'):
                 'is_shutdown': self.is_shutdown,
                 'is_ready': self.is_ready,
                 'cancel_event': self.cancel_event,
-                'timeout_ctrl': self._timeout_ctrl,
-                'zed_runtime_ctrl_address': self._zed_runtime_ctrl_address,
                 'runtime_cls': self.runtime_cls,
             },
         )
         self.daemon = self.args.daemon  #: required here to set process/thread daemon
 
     def _set_ctrl_adrr(self):
         """Sets control address for different runtimes"""
-        # This logic must be improved specially when it comes to naming. It is about relative local/remote position
-        # between the runtime and the `ZEDRuntime` it may control
-        from ..zmq import Zmqlet
-        from ..runtimes.container import ContainerRuntime
-
-        if self.runtime_cls == ContainerRuntime:
-            # Checks if caller (JinaD) has set `docker_kwargs['extra_hosts']` to __docker_host__.
-            # If yes, set host_ctrl to __docker_host__, else keep it as __default_host__
-            # Reset extra_hosts as that's set by default in ContainerRuntime
-            if (
-                self.args.docker_kwargs
-                and 'extra_hosts' in self.args.docker_kwargs
-                and __docker_host__ in self.args.docker_kwargs['extra_hosts']
-            ):
-                ctrl_host = __docker_host__
-                self.args.docker_kwargs.pop('extra_hosts')
-            else:
-                ctrl_host = self.args.host
-
-            self._zed_runtime_ctrl_address = Zmqlet.get_ctrl_address(
-                ctrl_host, self.args.port_ctrl, self.args.ctrl_with_ipc
-            )[0]
-        else:
-            self._zed_runtime_ctrl_address = Zmqlet.get_ctrl_address(
-                self.args.host, self.args.port_ctrl, self.args.ctrl_with_ipc
-            )[0]
+        self._zed_runtime_ctrl_address = self.runtime_cls.get_control_address(
```

Comment:
should this parameter be renamed to just `_runtime_ctrl_address`, since it is for different runtimes?

Reply:
This is on purpose .I want to make here clear that the control address is only valid for the zedruntime living anywhere (remote via jinad or via docker). The other Runtimes do not have control address that one cares about.Actually I think that ContainrrRuntime and JinaDRuntime should not exist at the same level as ZEDRuntime, I think that Pea should eventually just start extra processes to stream logs from docker and from jinad. But this is out of scope for now.

Reply:
>  ContainrrRuntime and JinaDRuntime should not exist at the same level as ZEDRuntime
This might sound a bit confusing for the developers especially when they are all named as `Runtime`